src/components/admin/ContentRenderer.tsx

```diff
@@ -64,52 +72,48 @@ export const getMetadata = async (contentId: number) => {
   //     segments: metadata['segments'],
   //   };
   // }
+
+  const bunnyUrls = {
+    1080: bunnyUrl(metadata[`video_1080p_mp4_${userId}`]),
+    720: bunnyUrl(metadata[`video_720p_mp4_${userId}`]),
+    360: bunnyUrl(metadata[`video_360p_mp4_${userId}`]),
+    subtitles: metadata['subtitles'],
+    slides: metadata['slides'],
+    segments: metadata['segments'],
+    thumbnails: metadata['thumbnail_mosiac_url'],
+  };
+
   if (user.bunnyProxyEnabled) {
-    return {
-      //@ts-ignore
-      1080: bunnyUrl(metadata[`video_1080p_mp4_${userId}`]),
-      //@ts-ignore
-      720: bunnyUrl(metadata[`video_720p_mp4_${userId}`]),
-      //@ts-ignore
-      360: bunnyUrl(metadata[`video_360p_mp4_${userId}`]),
-      subtitles: metadata['subtitles'],
-      //@ts-ignore
-      slides: metadata['slides'],
-      //@ts-ignore
-      segments: metadata['segments'],
-      // @ts-ignore
-      thumnnails: metadata['thumbnail_mosiac_url'],
-    };
+    return bunnyUrls;
   }
-  return {
-    //@ts-ignore
+
+  const mainUrls = {
     1080: metadata[`video_1080p_mp4_${userId}`],
-    //@ts-ignore
     720: metadata[`video_720p_mp4_${userId}`],
-    //@ts-ignore
     360: metadata[`video_360p_mp4_${userId}`],
     subtitles: metadata['subtitles'],
-    //@ts-ignore
     slides: metadata['slides'],
-    //@ts-ignore
     segments: metadata['segments'],
+    thumbnails: metadata['thumbnail_mosiac_url'],
   };
-  return {
-    //@ts-ignore
-    1080: metadata[`video_1080p_${userId}`],
-    //@ts-ignore
-    720: metadata[`video_720p_${userId}`],
-    //@ts-ignore
-    360: metadata[`video_360p_${userId}`],
-    //@ts-ignore
-    subtitles: metadata['subtitles'],
-    //@ts-ignore
-    slides: metadata['slides'],
-    //@ts-ignore
-    segments: metadata['segments'],
-    // @ts-ignore
-    thumnnails: metadata['thumbnail_mosiac_url'],
-  };
+
+  const isHighestQualityUrlAccessible = await isUrlAccessible(mainUrls['1080']);
+
+  if (isHighestQualityUrlAccessible) {
+    return mainUrls;
+  }
+
+  const otherQualities = ['720', '360'];
+  for (const quality of otherQualities) {
+    const urlKey = `${quality}`;
```

Comment:
urlKey = `${quality}` ?
Why not urlKey = quality?
Even better do u need urlKey?

Reply:
there was no need for the urlKey variable,  I have removed the urlKey variable altogether and directly used the quality value to access the property from the mainUrls object.